src/win32/posix_w32.c

```diff
@@ -390,12 +394,20 @@ int p_readlink(const char *path, char *buf, size_t bufsiz)
 	return (int)bufsiz;
 }
 
-int p_symlink(const char *old, const char *new)
+int p_symlink(const char *target, const char *path)
 {
-	/* Real symlinks on NTFS require admin privileges. Until this changes,
-	 * libgit2 just creates a text file with the link target in the contents.
-	 */
-	return git_futils_fake_symlink(old, new);
+	git_win32_path target_w, path_w;
+	wchar_t *target_p;
+
+	if (git_win32_path_from_utf8(path_w, path) < 0 ||
+		git__utf8_to_16(target_w, MAX_PATH, target) < 0)
+		return -1;
+
+	if (!CreateSymbolicLinkW(path_w, target_w,
+	    SYMBOLIC_LINK_FLAG_ALLOW_UNPRIVILEGED_CREATE))
+		return -1;
```

Comment:
Don't we have to handle the error case a bit more verbosely? I'm not really familiar with the win32 POSIX layer, but we usually invoke `set_errno()` in case where the call fails, don't we?

Reply:
We should be good here.  For the posix emulation functions, they can _generally_ fail with an error code without doing anything else.  Callers should then `giterr_set(GITERR_OS...` and `giterr_set` is smart enough to read the win32 system error message.

We only need to `set_errno` when callers actually read the `errno` back out and make a decision about how to handle the error.  (eg, looking for `EAGAIN`).  We try to carefully map the Windows failure into something that callers can use.  And we do it carefully to ensure that callers can understand that `errno` in the way they want to proceed for both Win32 and the actual POSIX.  Trying to make it general purpose instead would be 😓 .

Since we're not doing that here, we're okay to skip that step.